### DISCUSSION TITLE
Only show Credentials provider in the Sign in page

### DISCUSSION THREAD

**Discussion by zakaryan2004** [nextauthjs/next-auth#3133] [Help] [answered]
I have two providers: CredentialsProvider (for users) and GoogleProvider (for admins). I want to make a sign in button, which, when clicked, will only show the Credentials fields, and not the Google OAuth button. I have tried to use `signIn('credentials')`, but it seems that I need to pass the fields to the callback as well. I want the user to be redirected to the built-in signin page, but without the OAuth buttons. Is this possible with the built-in page? Thanks.

**Comment by balazsorban44** [comment, marked as answer]
Not by default, but I believe this should work:

```js
// /api/auth/[...nextauth].js
import NextAuth from "next-auth"
import CredentialsProvider from "next-auth/providers/credentials"
import GoogleProvider from "next-auth/providers/google"

export default async function auth(req, res) {
  const providers = [
    CredentialsProvider({
      // ...
    }),
    GoogleProvider({
      // ...
    })
  ]

  // Based on https://next-auth.js.org/getting-started/rest-api#get-apiauthsignin
  const isDefaultSigninPage = req.method === "GET" && req.query.nextauth.includes("signin")
  if (isDefaultSigninPage) providers.pop()

  return await NextAuth(req, res, {
    providers,
    // rest of your config ...
  })
}
```

So whenever the default sign-in page gets rendered, you just exclude the `GoogleProvider`.

**Reply by zakaryan2004** [reply]
Oh, I didn't realize I can do it that way. Thanks a lot, will check it!

**Reply by balazsorban44** [reply]
Do tell how it went!

**Reply by zakaryan2004** [reply]
It did work, thanks!

**Reply by zakaryan2004** [reply]
Just for anyone wondering, I needed to make a small modification so signin('google') would work.

`const isDefaultSigninPage = req.method === "GET" && req.query.nextauth.includes("signin") && !req.query.nextauth.includes("google")`

**Reply by balazsorban44** [reply]
Hmm. I wonder why the modification is needed. `signIn("google")` does a POST request, so the `req.method === "GET"` part would be enough.

**Reply by muckee** [reply]
This method fails using latest versions of Next and NextAuth and implementing the AppRouter pattern, as opposed to the PageRouter.

I've been unable to debug it so far, but I've [raised an issue](https://github.com/nextauthjs/next-auth/issues/7895), which includes a link to a [minimal repo](https://github.com/muckee/NextAuth-cannot-access-query-prop-of-res-object) to demonstrate the problem.

Just thought I should also mention it at 'the source'.

Would be good if the NextAuth docs were to imitate the Next.js docs and separate app and page routing under separate tabs, or something.

**Comment by balazsorban44** [comment]
I just referred to this discussion in a new section in the docs FYI :slightly_smiling_face:  https://github.com/nextauthjs/docs/pull/109

https://docs-git-init-instructions-nextauthjs.vercel.app/configuration/initialization#:~:text=The%20idea%20is%20taken%20from%20this%20discussion
